plugins/web-worker-offloading/hooks.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Hook callback for Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ * @package web-worker-offloading
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * PartyTown Configuration
+ *
+ * @since n.e.x.t
+ * @see https://partytown.builder.io/configuration
+ * @return array<string, mixed> Configuration for PartyTown.
+ */
+function wwo_configuration(): array {
+	$plugin_dir           = plugin_dir_path( __FILE__ );
+	$content_dir_basename = basename( WP_CONTENT_DIR );
+
+	$config = array(
+		'lib'     => substr( $plugin_dir, strpos( $plugin_dir, $content_dir_basename ) - 1 ) . 'build/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for PartyTown.
+	 *
+	 * @since n.e.x.t
+	 * @see <https://partytown.builder.io/configuration>.
+	 * @param array $config Configuration for PartyTown.
+	 * @return array
+	 */
+	return apply_filters( 'wwo_configuration', $config );
+}
+
+/**
+ * Initialize Web Worker Offloading.
+ *
+ * @since n.e.x.t
+ */
+function wwo_init(): void {
+	$partytown_js = file_get_contents( __DIR__ . '/build/partytown.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
+
+	if ( ! $partytown_js ) {
+		return;
+	}
+
+	wp_register_script(
+		'partytown',
+		'',
+		array(),
+		WEB_WORKER_OFFLOADING_VERSION,
+		array( 'in_footer' => false )
+	);
+
+	wp_add_inline_script(
+		'partytown',
+		sprintf(
+			'window.partytown = %s;',
+			wp_json_encode( wwo_configuration() )
+		),
+		'before'
+	);
+
+	wp_add_inline_script(
+		'partytown',
+		$partytown_js,
+		'after'
+	);
```

Comment:
Idea: Let's not reference `partytown` specifically. Let's call this script something like `web-worker-offloader`. That will keep us from having to be too bound up with the Partytown implementation.

Reply:
Done in https://github.com/WordPress/performance/pull/1247/commits/edaa30507ef2a9f7950d82cf8a6344e6a5e31a1a